### PR TITLE
Add dav1d/0.8.1

### LINKED ISSUE #4364: [request] dav1d - AV1 decoder
### Package Details
  * Package Name/Version: **dav1d/latest**
  * Website: **https://github.com/videolan/dav1d**
  * Source code: https://github.com/videolan/dav1d - https://code.videolan.org/videolan/dav1d

### Description Of The Library / Tool

> dav1d is a new AV1 cross-platform decoder, open-source, and focused on speed and correctness.

AV1 is an open codec from the [Alliance For Open Media](https://aomedia.org)

### PR DESCRIPTION
**Pull request by theirix**
Specify library name and version:  **dav1d/0.8.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Closes #4364

### CODE REVIEW

**Comment by SpaceIm** [comment]
LGTM. Could you explicitly set pkg_config in `package_info()` (even if default value is fine) please?

File: recipes/dav1d/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        "with_tools": [True, False],
+        "assembly": [True, False],
+        "with_avx512": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "bit_depth": "all",
+        "with_tools": True,
+        "assembly": False,
```
Comment (SpaceIm) [inline recipes/dav1d/all/conanfile.py:26]: shouldn't it be True by default? https://github.com/videolan/dav1d/blob/6ed5fafb42c651c24b6a65fd4f50ed426fd72d65/meson_options.txt#L10

File: recipes/dav1d/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+        if self.settings.compiler == "Visual Studio" and not self.options.shared:
+            # https://github.com/mesonbuild/meson/issues/7378
+            os.rename(os.path.join(self.package_folder, "lib", "libdav1d.a"),
+                      os.path.join(self.package_folder, "lib", "dav1d.lib"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["dav1d"]
+        if self.settings.os == "Linux":
+            self.cpp_info.system_libs.extend(["dl", "pthread"])
```
Comment (SpaceIm) [inline recipes/dav1d/all/conanfile.py:104]:
```suggestion
            self.cpp_info.system_libs.extend(["dl", "pthread"])
        
        if self.options.with_tools:
            bin_path = os.path.join(self.package_folder, "bin")
            self.output.info("Appending PATH environment variable: {}".format(bin_path))
            self.env_info.PATH.append(bin_path)
```

File: recipes/dav1d/all/conanfile.py (hunk 1 above)
Comment (theirix) [reply recipes/dav1d/all/conanfile.py:26]: I prefer to disable options that make non-portable binaries that can build on CI but would not work on client machines without rebuilding. For this library, we can enable assembly by default (I found no autodetect) while keeping avx512 off. How do you think?

Comment (SpaceIm) [reply recipes/dav1d/all/conanfile.py:26]: Agree, exactly what I was thinking 😉

Comment (theirix) [reply recipes/dav1d/all/conanfile.py]: Done

**Review by SpaceIm — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
bot failed to push two updates to GitHub 😖

**Comment by prince-chrismc** [comment]
🤞

**Comment by theirix** [comment]
Very strange thing. Reproduced the build (debug with assembly) locally. The reason for the failing build is passing `CFLAGS='-m64 -g -fPIC'` from the environment. Build succeeds when I remove flag '-g'.

**Comment by prince-chrismc** [comment]
#4420 ?

**Comment by Croydon** [comment]
Could you provide a reference for the bug, so that we have something to track and know when meson can be updated?

**Comment by theirix** [comment]
> Could you provide a reference for the bug, so that we have something to track and know when meson can be updated?
Sure, it is in a commit message:

Reproduced on debug builds when Conan passes additional '-g' flag to CFLAGS.
Meson fails to drop duplicate '-g' flags and the dav1d check
'cc.symbols_have_underscore_prefix' fails which leads to undefined
symbols while linking.

Seems like bug introduced after 0.51.0, upstream ref mesonbuild/meson#5482
Correct behaviour can be checked by locating string '%define PREFIX 1' in
the generated config.asm.

**Comment by theirix** [comment]
@SpaceIm unfortunately, Windows build with assembler is very flaky so I disabled it. Reproduced one problem locally - linker just hangs. Another problem is LNK1000 on CI. I found several reports about linker hangs but no clues on fixing them.

File: recipes/dav1d/all/conanfile.py [hunk 3]
```diff
@@ -52,7 +55,10 @@ def configure(self):
         del self.settings.compiler.cppstd
 
     def build_requirements(self):
-        self.build_requires("meson/0.56.2")
+        # Meson versionssince 0.51.0 do not work with autodetect symbol prefix
```
Comment (prince-chrismc) [inline recipes/dav1d/all/conanfile.py:58]:
```suggestion
        # Meson versions since 0.51.0 do not work with autodetect symbol prefix
```

**Review by prince-chrismc — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Pull request merged**